### PR TITLE
Support valid data for multitask training

### PR DESCRIPTION
**Pull request by zjgemi**
## Summary by CodeRabbit

- **New Features**
  - Introduced support for multitask validation data, allowing users to specify multiple validation datasets through new arguments.
  - Enhanced flexibility in handling validation data, accommodating both single and multitask configurations.
  - Added dynamic model freezing capability based on configuration parameters.
  - Improved configurability by allowing external configuration data to be integrated into function execution.

- **Bug Fixes**
  - Improved error handling for validation data inputs, ensuring robust processing of various data structures.

- **Documentation**
  - Updated documentation to clarify the usage of new arguments related to multitask validation data.

### CODE REVIEW

File: dpgen2/op/run_relax.py [hunk 1]
```diff
@@ -73,6 +81,35 @@ def execute(
         task_group = ip["diffcsp_task_grp"]
         task = next(iter(task_group))  # Only support single task
         models = ip["models"]
+        if ip["expl_config"].get("head") is not None:
```
Comment (wanghan-iapcm) [inline dpgen2/op/run_relax.py:84]: something like `model_frozen_head` may be more self-explanatory.

Comment (wanghan-iapcm) [reply dpgen2/op/run_relax.py:84]: sry this suggestion should have been raised for run_lmp

File: dpgen2/op/run_relax.py [hunk 2]
```diff
@@ -73,6 +81,35 @@ def execute(
...
+                                out,
+                                "\n",
+                                "err msg",
+                                err,
+                                "\n",
+                            )
+                        )
+                    )
+                    raise RuntimeError("freeze failed")
+                frozen_models.append(Path(mname))
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply dpgen2/op/run_relax.py:111]: @zjgemi I also find the suggestion by coderabbit is actually a good idea. the `freeze_model` function can be shared among run_lmp and run_relax.

File: dpgen2/op/run_lmp.py [hunk 3]
```diff
@@ -240,7 +219,9 @@ def lmp_args():
                 default=False,
                 doc=doc_shuffle_models,
             ),
-            Argument("head", str, optional=True, default=None, doc=doc_head),
+            Argument(
+                "model_frozen_head", str, optional=True, default=None, doc=doc_head
```
Comment (wanghan-iapcm) [inline dpgen2/op/run_lmp.py:223]:
```suggestion
                "model_frozen_head", str, optional=True, default=None, doc=doc_head, alias=["head"],
```

**Review by wanghan-iapcm — COMMENTED** [review]
You may want to provide config argument check for the `run_relax`.

File: dpgen2/op/run_lmp.py (hunk 3 above)
Comment (zjgemi) [reply dpgen2/op/run_lmp.py:223]: This parameter is internal and passed within dpgen2 rather than for users. The name here is just for better clarity.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**